src/lgdo/lh5/_serializers/read/utils.py

```diff
@@ -94,3 +101,69 @@ def read_n_rows(h5o, fname, oname):
 
     msg = f"don't know how to read rows of LGDO {lgdotype.__name__}"
     raise LH5DecodeError(msg, fname, oname)
+
+
+def read_size_in_bytes(h5o, fname, oname, field_mask=None):
+    """Read number of rows in LH5 object"""
+    if not h5py.h5a.exists(h5o, b"datatype"):
+        msg = "missing 'datatype' attribute"
+        raise LH5DecodeError(msg, fname, oname)
+
+    h5a = h5py.h5a.open(h5o, b"datatype")
+    type_attr = np.empty((), h5a.dtype)
+    h5a.read(type_attr)
+    type_attr = type_attr.item().decode()
+    lgdotype = datatype.datatype(type_attr)
+
+    # scalars are dim-0 datasets
+    if lgdotype in (types.Scalar, types.Array, types.ArrayOfEqualSizedArrays):
+        return int(np.prod(h5o.shape) * h5o.dtype.itemsize)
+
+    # structs don't have rows
+    if lgdotype is types.Struct:
```

Comment:
this does not include Histograms, but it should (i.e. `is` checks for type _equality_, not _inheritance_)?

`FixedSizeArray` is also missing.

both also applies to `read_n_rows` above, I suppose?